popcorn/node.rb

```diff
@@ -0,0 +1,12 @@
+# Represents a node inside a graph
+class Node
+  attr_reader :name, :adjacent
+
+  def initialize(name)
+    @name = name
+  end
+
+  def assign_adjacency(nodes)
```

Comment:
I agree and i have thought about it but it is *the chicken or the egg* problem here. I am not sure if there is a way to create the nodes and assign adjacency at the same time. Let's say we have node A *linked* to node B. How would we code it  ```a_node = Node.new('A', [Node.new('B', ??]) .  ```

Reply:
It could be implemented with different architecture. Have a `Graph` class that is constructed with a collection of vertices and adjacency list/matrix or collection of edges. This way a vertex object knows only about its label and adjacency objects know about connections and weights